### PR TITLE
Port workflow directory from master

### PR DESCRIPTION
**Pull request by uzust01**
Add current Zowe-master-workflows code to lts-incremental.
Includes: #249 #254 #257 #258 #262 #263

### CODE REVIEW

**Comment by ChrisBoehmCA** [comment]
Thanks very much for porting the recent workflow changes. It looks like a snapshot needs to be updated

File: packages/workflows/src/cli/create/dataset/Dataset.definition.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { join } from "path";
 export const DataSet: ICommandDefinition = {
     name: "workflow-from-data-set",
     aliases: ["wfds"],
-    description: "Create a workflow instance in z/OSMF using a Data set",
+    description: "Create a z/OMF workflow on a z/OS system using a Data set",
```
Comment (zFernand0) [inline packages/workflows/src/cli/create/dataset/Dataset.definition.ts]: shouldn't it be `z/OSMF`?

File: packages/workflows/src/api/WorkflowValidator.ts [hunk 2]
```diff
@@ -68,42 +68,6 @@ export class WorkflowValidator {
...
 
-    /**
-     * Validate supplied parameter
-     * @static
-     * @param {string} path - string to check if contains USS path or DSNAME
-     * @param {string} errorMsg - message to show in case validation fails
-     * @memberof WorkflowValidator
-     * TODO maybe also validation if the file or DSNAME exists on Mainframe
-     */
-    public static validatePath(path: string, errorMsg: string) {
```
Comment (zFernand0) [inline packages/workflows/src/api/WorkflowValidator.ts]: I get it that people often do not use the CLI APIs that we provide, but we should be more careful when removing functionality from the project.

**Comment by zFernand0** [comment]
I marked this as a "minor" enhancement for now.
If someone else has a concern that this is potentially a breaking PR, then changes will be requested.

File: packages/workflows/src/cli/create/dataset/Dataset.definition.ts (hunk 1 above)
Comment (uzust01) [reply packages/workflows/src/cli/create/dataset/Dataset.definition.ts]: Yes, it should be, thanks 🤖

**Comment by ChrisBoehmCA** [comment]
I think it's not breaking. The removed API was an internal input validation API, so I think it's okay

**Comment by zFernand0** [comment]
> I think it's not breaking. The removed API was an internal input validation API, so I think it's okay

Thank you for pointing out that the validation APIs were never exported.
https://github.com/muzikovam/zowe-cli/blob/c09a691fc7ec37339f3c5a6b63473a8fcafc007b/packages/workflows/index.ts#L22-L29

File: packages/workflows/src/api/WorkflowValidator.ts (hunk 2 above)
Comment (zFernand0) [reply packages/workflows/src/api/WorkflowValidator.ts]: Validation APIs were never exported

**Pull request merged**